### ISSUE TITLE
3.0 Deprecation Warning: Deprecate fallback spaCy model

### ISSUE THREAD

**Issue by ancalita** [RasaHQ/rasa#8882]
**Background**
This warning was found in:
- rasa.nlu.utils.spacy_utils.py:L140
```
"f""SpaCy model is not properly configured! ""
 f""Please add a `model` property to `SpacyNLP`. ""
f""Will use '{spacy_model_name}' as a fallback spaCy model. ""
f""This fallback will be deprecated in Rasa 3.0"""
```

**Definition of Done**
- [x] resolve warning in 3.0 version
- [x] Removed all uses of _check_model_fallback in favour of using directly model from config. Here I assume we don't have to do any checking on model -- if a Spacy model with that name doesn't exist, then Spacy will raise an error? 🤔 Currently, we do checking but it requires hard-coding all possible Spacy model names, which then leads to issues like Support spaCy 3.1 #9068 which are just about updating that hard-coded list afaics.
- [x] Remove tests for _check_model_fallback
- [x] Removed the references in the docs

**Comment by koaning** [comment]
I might be able to pick this one up. I implemented the spaCy 3.0 and wouldn't mind wrapping it up for Rasa 3.0. 

Just to check, I can make a fix with a PR to master?

**Comment by samsucik** [comment]
@koaning this issue needs to be done before 3.0, while #9068 doesn't necessarily have to go into 3.0 (strictly speaking). Seeing that you assigned yourself on this one, are you actually working on it? To me it looks like an issue that research can easily pick up...

As for a definition of done, the method in question (`_check_model_fallback`) is found in the codebase and in some tests (but not in docs), so we might say:
1. Removed all uses of `_check_model_fallback` in favour of using directly `model` from config. _Here I assume we don't have to do any checking on `model` -- if a Spacy model with that name doesn't exist, then Spacy will raise an error? 🤔  Currently, we [do checking but it requires hard-coding all possible Spacy model names](https://github.com/RasaHQ/rasa/blob/300df5ac085c174521599364efabddb4000b6716/rasa/nlu/utils/spacy_utils.py#L105), which then leads to issues like #9068 which are just about updating that hard-coded list afaics._
2. Remove tests for `_check_model_fallback`

**Comment by koaning** [comment]
@samsucik I'll gladly pick it up if it removes a burden from the research team. If you folks have the capacity then I certainly won't mind anybody else picking this up. 👍 

I agree that strictly speaking it's no needed to upgrade spaCy. There are some extra languages added to the newer version though, which would be nice to add.

**Comment by TyDunn** [comment]
@koaning Thanks for picking this up. This will be blocking the architecture revamp. Once 2.8 is released early next week, if you have not completed this issue, Enable might reach out about taking it over, so it is not blocking them

**Comment by koaning** [comment]
Clear. I've set it as a TODO for Monday.

**Comment by koaning** [comment]
The PR is ready and no longer a work in progress. The upgrade to spaCy 3.1 will be a separate PR.
